Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/TempSpawnableAssetsCache.cpp

```diff
@@ -0,0 +1,225 @@
+        if (!templateReference.has_value())
+        {
+            return AZ::Failure(AZStd::string::format("Could not get Template DOM given Template's id %llu .", templateId));
+        }
+
+        // Use a random uuid as this is only a temporary source.
+        PrefabConversionUtils::PrefabProcessorContext context(AZ::Uuid::CreateRandom());
+        PrefabDom copy;
+        copy.CopyFrom(templateReference->get().GetPrefabDom(), copy.GetAllocator(), false);
+        context.AddPrefab(spawnableName, AZStd::move(copy));
+        m_converter.ProcessPrefab(context);
+
+        if (!context.HasCompletedSuccessfully() || context.GetProcessedObjects().empty())
+        {
+            return AZ::Failure(AZStd::string(
+                "Failed to convert the prefab into assets."
+                "Please confirm that the 'TempSpawnables' prefab processor stack is capable of producing a usable product asset."));
+        }
+
+        static constexpr size_t NoTargetSpawnable = AZStd::numeric_limits<size_t>::max();
+        size_t targetSpawnableIndex = NoTargetSpawnable;
+        AZStd::vector<AZ::Data::AssetId> assetIds;
+
+        // Create temporary assets from the processed data.
+        for (auto& product : context.GetProcessedObjects())
+        {
+            if (product.GetAssetType() == AZ::AzTypeInfo<AzFramework::Spawnable>::Uuid() &&
+                product.GetId() ==
+                AZStd::string::format("%s.%s", spawnableName.data(), AzFramework::Spawnable::FileExtension))
+            {
+                targetSpawnableIndex = m_assets.size();
+            }
+
+            AZ::Data::AssetInfo info;
+            info.m_assetId = product.GetAsset().GetId();
+            info.m_assetType = product.GetAssetType();
+            info.m_relativePath = product.GetId();
+
+            AZ::Data::AssetCatalogRequestBus::Broadcast(
+                &AZ::Data::AssetCatalogRequestBus::Events::RegisterAsset, info.m_assetId, info);
+            m_assets.emplace_back(product.ReleaseAsset().release(), AZ::Data::AssetLoadBehavior::Default);
+
+            // Ensure the product asset is registered with the AssetManager
+            // Hold on to the returned asset to keep ref count alive until we assign it the latest data
+            AZ::Data::Asset<AZ::Data::AssetData> asset =
+                AZ::Data::AssetManager::Instance().FindOrCreateAsset(info.m_assetId, info.m_assetType, AZ::Data::AssetLoadBehavior::Default);
+
+            // Update the asset registered in the AssetManager with the data of our product from the Prefab Processor
+            AZ::Data::AssetManager::Instance().AssignAssetData(m_assets.back());
+        }
+
+        if (targetSpawnableIndex == NoTargetSpawnable)
+        {
+            return AZ::Failure(AZStd::string("Failed to produce the target spawnable '%.*s'", AZ_STRING_ARG(spawnableName)));
+        }
+        
+        if (loadCreatedAssets)
```

Comment:
`loadCreatedAssets` suggests that spawnables need to be loaded still, but since they're constructed in memory they're already fully formed. Perhaps a better name is `loadDependedAssets` or `loadReferencedAssets`?